CMakeLists.txt

```diff
@@ -1,52 +1,65 @@
+  GIT_REPOSITORY "${torch_fetch_git_repository}"
+  GIT_TAG        "${torch_fetch_git_tag}"
 )
+message(STATUS "FetchContent_Populate ${torch_fetch_git_repository} ${torch_fetch_git_tag}")
 FetchContent_Populate(torch_fetch)
+message(STATUS "FetchContent_Populate ${torch_fetch_git_repository} ${torch_fetch_git_tag} - done")
+
+# Use torch's setup.py to build and locally install it.
+if(NOT DEFINED torch_fetch_install_directory)
+  # We can enable and disable functionality with the following options.
+  if(NOT DEFINED ENV{USE_CUDA})
+    # We will not use CUDA on GitHub
+    set(ENV{USE_CUDA} "0")
+    set(torch_fetch_additional_libraries "torch_cuda" "c10_cuda")
+  endif()
+  if(NOT DEFINED ENV{USE_CUDNN})
+    # We will not use CUDNN on GitHub
+    set(ENV{USE_CUDNN} "0")
+  endif()
+  set(torch_fetch_install_directory "${torch_fetch_SOURCE_DIR}/../torch_fetch-install")
+  set(torch_fetch_build_directory "${torch_fetch_SOURCE_DIR}/build")
+  message(STATUS "Executing: cd ${torch_fetch_SOURCE_DIR}; ${Python_EXECUTABLE} setup.py install --prefix=${torch_fetch_install_directory}")
+  add_custom_target(
+    temp ALL
+    ${Python_EXECUTABLE} setup.py install --prefix=${torch_fetch_install_directory}
+    WORKING_DIRECTORY ${torch_fetch_SOURCE_DIR}
+    )
+  message(STATUS "Executing: cd ${torch_fetch_SOURCE_DIR}; ${Python_EXECUTABLE} setup.py install --prefix=${torch_fetch_install_directory} - done")
+endif()
+
+include_directories(
+  "${torch_fetch_install_directory}/torch/include/torch/csrc/api/include"
+  "${torch_fetch_install_directory}/torch/include"
+  "${torch_fetch_install_directory}/caffe2/aten/src"
+  "${torch_fetch_install_directory}/lib/python3.8/site-packages/torch/include"
+  "${torch_fetch_build_directory}/aten/src"
+  "${torch_fetch_build_directory}/nccl/include"
+  "${torch_fetch_SOURCE_DIR}/aten/src"
+  )
+
+add_subdirectory(${torch_fetch_SOURCE_DIR} ${torch_fetch_build_directory})
 
-# get_cmake_property(_variableNames VARIABLES)
-# list (SORT _variableNames)
-# foreach (_variableName ${_variableNames})
-#     message(STATUS "${_variableName}=${${_variableName}}")
-# endforeach()
-
-set(Torch_DIR "${torch_fetch_SOURCE_DIR}/share/cmake/Torch")
-find_package(Torch REQUIRED CONFIG)
-
-# set(_Torch_lib Torch::Torch)
-# set(Torch_DIR_INSTALL ${Torch_DIR})
-# set(Torch_DIR_BUILD ${Torch_DIR})
-# # It is only needed to EXPORT_CODE_BUILD when using external Torch
-# set(${PROJECT_NAME}_EXPORT_CODE_BUILD "${${PROJECT_NAME}_EXPORT_CODE_BUILD}
-# if(NOT ITK_BINARY_DIR)
-#   set(Torch_DIR \"${Torch_DIR_BUILD}\")
-#   find_package(Torch REQUIRED CONFIG)
-# endif()
-# ")
-#
-# include_directories(${TORCH_INCLUDE_DIRS})
-# # itk_module_target(${_Torch_lib} NO_INSTALL)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS}")
-set(PyTorch_LIBRARIES PyTorch "${TORCH_LIBRARIES}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,--no-as-needed")
+list(APPEND PyTorch_LIBRARIES "${TORCH_LIBRARIES}" "${torch_fetch_additional_libraries}")
```

Comment:
Where does `TORCH_LIBRARIES` come from? Maybe `TORCH_INCLUDE_DIRS` is also provided, so we can avoid manual list on line 50.

Reply:
TORCH_LIBRARIES doesn't exists.  Removing it.